sources/cassandra/infoschema.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gocql/gocql"
+
+	sp "cloud.google.com/go/spanner"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/schema"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/sources/common"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/spanner/ddl"
+)
+
+// Connect establishes a session with the Cassandra cluster using parameters from SourceProfile.
+func Connect(sourceProfile profiles.SourceProfile) (SessionInterface, error) {
+	cfg := sourceProfile.Conn.Cassandra
+
+	contactPoint := strings.TrimSpace(cfg.Host)
+
+	cluster := gocql.NewCluster(contactPoint)
```

Comment:
Adding an accessor for `gocql` might help in UTing most of the code under review (though not 100%)